src/wasm-lib/kcl/src/lint/checks/offset_plane.rs

```diff
@@ -0,0 +1,241 @@
+        return Ok(vec![]);
+    };
+
+    let Expr::ObjectExpression(ref plane) = plane.inner.value else {
+        return Ok(vec![]);
+    };
+
+    let mut origin: Option<(f64, f64, f64)> = None;
+    let mut x_vec: Option<(f64, f64, f64)> = None;
+    let mut y_vec: Option<(f64, f64, f64)> = None;
+    // let mut z_vec: Option<(f64, f64, f64)> = None;
+
+    for property in &plane.inner.properties {
+        let Expr::ObjectExpression(ref point) = property.inner.value else {
+            return Ok(vec![]);
+        };
+
+        let Some((x, y, z)) = get_xyz(&point.inner) else {
+            return Ok(vec![]);
+        };
+
+        let property_name = &property.inner.key.inner.name;
+
+        match property_name.as_str() {
+            "origin" => origin = Some((x, y, z)),
+            "xAxis" => x_vec = Some((x, y, z)),
+            "yAxis" => y_vec = Some((x, y, z)),
+            // "zAxis" => z_vec = Some((x, y, z)),
+            _ => {
+                continue;
+            }
+        };
+    }
+
+    let Some(origin) = origin else { return Ok(vec![]) };
+    let Some(x_vec) = x_vec else { return Ok(vec![]) };
+    let Some(y_vec) = y_vec else { return Ok(vec![]) };
+    // let Some(z_vec) = z_vec else { return Ok(vec![]) };
+
+    if [origin.0, origin.1, origin.2].iter().filter(|v| **v == 0.0).count() < 2 {
+        return Ok(vec![]);
+    }
+    // two of the origin values are 0, 0; let's work it out and check
+    // what's **up**
+
+    /// This will attempt to very poorly translate orientation to a letter
+    /// if it's possible to do so. The engine will norm these vectors, so
+    /// we'll just use logic off 0 for now, but this sucks, generally speaking.
+    fn vector_to_letter<'a>(x: f64, y: f64, z: f64) -> Option<&'a str> {
+        if x > 0.0 && y == 0.0 && z == 0.0 {
+            return Some("X");
+        }
+        if x < 0.0 && y == 0.0 && z == 0.0 {
+            return Some("-X");
+        }
+
+        if x == 0.0 && y > 0.0 && z == 0.0 {
+            return Some("Y");
+        }
+        if x == 0.0 && y < 0.0 && z == 0.0 {
+            return Some("-Y");
+        }
+
+        if x == 0.0 && y == 0.0 && z > 0.0 {
+            return Some("Z");
+        }
+        if x == 0.0 && y == 0.0 && z < 0.0 {
+            return Some("-Z");
+        }
+
+        None
+    }
+
+    let allowed_planes = HashMap::from([
+        // allowed built-in planes
+        ("XY".to_owned(), true),
+        ("-XY".to_owned(), true),
+        ("XZ".to_owned(), true),
+        ("-XZ".to_owned(), true),
+        ("YZ".to_owned(), true),
+        ("-YZ".to_owned(), true),
+    ]);
+    // Currently, the engine **ONLY** accepts[1] the following:
+    //
+    // XY
+    // -XY
+    // XZ
+    // -XZ
+    // YZ
+    // -YZ
+    //
+    // [1]: https://zoo.dev/docs/kcl/types/PlaneData
+
+    let plane_name = format!(
+        "{}{}",
+        vector_to_letter(x_vec.0, x_vec.1, x_vec.2).unwrap_or(""),
```

Comment:
If `vector_to_letter` returns None, this lint should probably just fail right? There's no point silently returning just "Y" as a plane. Shouldn't this just be `Ok(vec![])` in that case? Or at least a debug_assertion that will panic?

Reply:
That's OK since we check it in the hash map, and checking each Option and returning if None is more work than its worth here. We can change that if you feel that strongly but it's not a huge deal because of the explicit plane check below